### ISSUE TITLE
bị lỗi khi tạo file audio có nhiều hơn 1 Text batch

### ISSUE THREAD

**Issue by dieptran2500** [EraX-JS-Company/EraX-Smile-F5TTS#5]
Khi mình tạo file audio ngắn, nằm trong 1 Text batch  thì file audio bình thường

với đoạn audio dài, số Text batch được tự động split ra > 1, thì file audio không sử dụng được

Mong hỗ trợ

Mình sử dụng máy window

Có 1 thay đổi nhỏ, do window không sử dụng vinorm được, nên mình đổi qua sử dụng vi_cleaner  để đổi mấy cái số thành chữ.

các thư viện được cài giống như hướng dẫn

accelerate==1.5.2
aiofiles==23.2.1
aiohappyeyeballs==2.6.1
aiohttp==3.11.14
aiosignal==1.3.2
annotated-types==0.7.0
antlr4-python3-runtime==4.9.3
anyio==4.9.0
asttokens==3.0.0
async-timeout==5.0.1
attrs==25.3.0
audioread==3.0.1
bitsandbytes==0.45.4
boto3==1.37.23
botocore==1.37.23
cached_path==1.7.1
cachetools==5.5.2
certifi==2025.1.31
cffi==1.17.1
charset-normalizer==3.4.1
click==8.1.8
colorama==0.4.6
comm==0.2.2
contourpy==1.3.1
cycler==0.12.1
datasets==3.5.0
debugpy==1.8.13
decorator==5.2.1
dill==0.3.8
docker-pycreds==0.4.0
einops==0.8.1
einx==0.3.0
ema-pytorch==0.7.7
encodec==0.1.1
exceptiongroup==1.2.2
executing==2.2.0
f5-tts==1.0.8
fastapi==0.115.12
ffmpy==0.5.0
filelock==3.18.0
fonttools==4.56.0
frozendict==2.4.6
frozenlist==1.5.0
fsspec==2024.12.0
gitdb==4.0.12
GitPython==3.1.44
google-api-core==2.24.2
google-auth==2.38.0
google-cloud-core==2.4.3
google-cloud-storage==2.19.0
google-crc32c==1.7.1
google-resumable-media==2.7.2
googleapis-common-protos==1.69.2
gradio==5.14.0
gradio_client==1.7.0
h11==0.14.0
httpcore==1.0.7
httpx==0.28.1
huggingface-hub==0.27.1
hydra-core==1.3.2
idna==3.10
ipykernel==6.29.5
ipython==8.34.0
jedi==0.19.2
jieba==0.42.1
Jinja2==3.1.6
jmespath==1.0.1
joblib==1.4.2
jupyter_client==8.6.3
jupyter_core==5.7.2
kiwisolver==1.4.8
lazy_loader==0.4
librosa==0.11.0
llvmlite==0.44.0
loguru==0.7.3
markdown-it-py==3.0.0
MarkupSafe==2.1.5
matplotlib==3.10.1
matplotlib-inline==0.1.7
mdurl==0.1.2
mpmath==1.3.0
msgpack==1.1.0
multidict==6.2.0
multiprocess==0.70.16
nest-asyncio==1.6.0
networkx==3.4.2
numba==0.61.0
numpy==1.24.0
omegaconf==2.3.0
orjson==3.10.16
packaging==24.2
pandas==2.2.3
parso==0.8.4
pillow==11.1.0
platformdirs==4.3.7
pooch==1.8.2
prompt_toolkit==3.0.50
propcache==0.3.1
proto-plus==1.26.1
protobuf==5.29.4
psutil==7.0.0
pure_eval==0.2.3
pyarrow==19.0.1
pyasn1==0.6.1
pyasn1_modules==0.4.2
pycparser==2.22
pydantic==2.11.1
pydantic_core==2.33.0
pydub==0.25.1
Pygments==2.19.1
pyparsing==3.2.3
pypinyin==0.54.0
python-dateutil==2.9.0.post0
python-multipart==0.0.20
pytz==2025.2
pywin32==310
PyYAML==6.0.2
pyzmq==26.3.0
regex==2024.11.6
requests==2.32.3
rich==13.9.4
rsa==4.9
ruff==0.11.2
s3transfer==0.11.4
safehttpx==0.1.6
safetensors==0.5.3
scikit-learn==1.6.1
scipy==1.15.2
semantic-version==2.10.0
sentry-sdk==2.24.1
setproctitle==1.3.5
shellingham==1.5.4
six==1.17.0
smmap==5.0.2
sniffio==1.3.1
soundfile==0.13.1
soxr==0.5.0.post1
stack-data==0.6.3
starlette==0.46.1
sympy==1.13.1
threadpoolctl==3.6.0
tokenizers==0.21.1
tomli==2.2.1
tomlkit==0.13.2
torch==2.6.0+cu126
torchaudio==2.6.0
torchdiffeq==0.2.5
torchvision==0.21.0+cu126
tornado==6.4.2
tqdm==4.67.1
traitlets==5.14.3
transformers==4.50.3
transformers-stream-generator==0.0.5
typer==0.15.2
typing-inspection==0.4.0
typing_extensions==4.13.0
tzdata==2025.2
urllib3==2.3.0
uvicorn==0.34.0
vi_cleaner @ git+https://github.com/CodeLinkIO/Vietnamese-text-normalization.git@4f7b9ea525de73b95838700abab33b68a96bfe85
vietnam_number @ git+https://github.com/CodeLinkIO/vietnam-number.git@2969648536e8074269f412643996028ae61742a8
vinorm==2.0.7
vocos==0.1.0
wandb==0.19.8
wcwidth==0.2.13
websockets==14.2
win32_setctime==1.2.0
x-transformers==2.1.37
xxhash==3.5.0
yarl==1.18.3

**Comment by EraX-AI** [comment]
Sao lại không nhỉ ? Sorry bận quá nên trả lời chậm. Để mình xem. Có thể có ký tự lạ nằm ngoài vocab.txt, bạn kiểm tra thử nhé.

**Comment by v-nhandt21** [comment]
Chào mọi người, xin lỗi mọi người vì các hạn chế của Vinorm với window và mac, mình có public source code vinorm ở đây, mọi người xem cùng contribute nha:
- https://github.com/v-nhandt21/vinorm_cpp_version
- Hồi trước mình code kiểu đọc ghi qua file sau đó wrap lại thành python nên nó không multiprocess được
- Để hỗ trợ thêm window thì cần phải build thêm ICU cho window
- Để update cho mac version các bạn tham khảo ở đây: https://github.com/v-nhandt21/vinorm_cpp_version/tree/mac-version

**Comment by dieptran2500** [comment]
> Chào mọi người, xin lỗi mọi người vì các hạn chế của Vinorm với window và mac, mình có public source code vinorm ở đây, mọi người xem cùng contribute nha:
> 
> * https://github.com/v-nhandt21/vinorm_cpp_version
> * Hồi trước mình code kiểu đọc ghi qua file sau đó wrap lại thành python nên nó không multiprocess được
> * Để hỗ trợ thêm window thì cần phải build thêm ICU cho window
> * Để update cho mac version các bạn tham khảo ở đây: https://github.com/v-nhandt21/vinorm_cpp_version/tree/mac-version

Cảm ơn bạn nhiều, có mã nguồn có thể chỉnh sửa và reimplement lại
